app/models/survey.rb

```diff
@@ -46,4 +51,22 @@ def questions
 
     Question.order("RANDOM()") + demographic_questions + customised_questions
   end
+
+  def self.master_survey
+    find_by_master(true)
+  end
+
+  protected
+
+  def only_one_master_survey
+    return unless master?
+
+    matches = Survey.master
+    if persisted?
+      matches = matches.where('id != ?', id)
+    end
+    if matches.exists?
+      errors.add(:master, "Can only have one master survey")
+    end
+  end
```

Comment:
@drewpotter if you read a little further down on the stack overflow post there is a much more elegant solution